### DISCUSSION TITLE
Akash Hackathon Proposal

### DISCUSSION THREAD

**Discussion by CaptainSplince** [akash-network/community#100] [Governance Proposals]
### Overview
With the launch of the new Akash website, open-sourcing the network and Akash’s new operational framework of SIGs, WGs and the steering committee, I believe it’s worthwhile hosting a marketing campaign around and how developers can get involved with the Akash ecosystem. I believe the best way to bring attention to what Akash is building at this juncture is hosting the first ever Akash Network hackathon.

### Why Host an Akash Hackathon?
Crypto is a loud space. There are hundreds of projects trying to get developers' attention. Looking at well funded projects, such as [Ethereum](https://ethglobal.com/), [Aptos](https://aptosfoundation.org/currents/aptos-world-tour-hackathon-coming-2023), [Near](https://near.org/metabuild/), [Juno](https://securesecrets-org.medium.com/hack-juno-create-build-earn-d91ab527c35), ect. It becomes plain to see that hackathons are a key driver of developer growth within a given crypto project. 

A hackathon will allow our marketing efforts to align on a number of important fronts.
- Bring attention to Akash and it’s technology
- Encourage contributions on Github
- Support development new ecosystem partners
- Partner and cross promote Akash across the Cosmos ecosystem

### How did we get here?
In late January 2023 the community formed a [Hackathon Working group](https://discord.gg/GPRryC88SA) where members of the Overclock Labs team, Akash Insiders and community members developed a [hackathon strategy](https://docs.google.com/document/d/1AE0k3QdejTY-s3B7zTZuCP06A5ie8n_yXF_lZGmUKUc/edit). 

[After meeting with community members](https://drive.google.com/file/d/1kSweMlwuQi1q84mZ52Tw_SDLQ1rJJALn/view?usp=sharing), feedback was included in the initial proposal doc and finalized for this proposal.

### Projected Outcomes

Because this is the first Akash Hackathon, it's hard to forecast outcomes. However, given audience size and current tools available on Akash, the below is an estimate on what can be expected from an Akash Hackathon

- Host a month long virtual hackathon driving 500 new contributions to the Akash github
- Onboard 100 new Akash EDU students
- Onboard 10 new Akash Insiders
- Onboard 2 new ecosystem partners (ex - Cloudmos, Terraform, Praetor)
- Drive 100 new deployments on the network
- Closer alignment with Cosmos ecosystem through sponsored tracks

### Timeline

Below is a timeline of activities in lead up to, during and after the proposed hackathon. Exact dates are yet to be determined until approved through community governance.

_1 month (Preparation)_
- Develop marketing assets
- Outreach and partnership with ecosystem players
- Develop Hackathon tracks
- Develop sponsored tracks
- Define hackathon timeline
- Define budget 
- Judge & mentor selection
- Judging criteria

_2 weeks (Registration)_
- Prepare final details of hackathon
- Encourage signups

_1 month (Execution)_
- Weekly mentor sessions
- Dedicated website
- Dedicated discord channel
- Demo Day (top 3 pitches w/judges)
- Review projects and select winners

_1 month (Review)_
- Feedback forms
- Announce winners
- Post mortem with organizers & sponsors

### Hackathon Roles (Support)
- _Hackathon Lead_ - Keeps trains running on time (operations and logistics)
- _Hackathon Outreach Manager_ - Support with getting sponsor tracks and mentors/judges
- _Hackathon Event Manager_ - Hosting weekly mentor sessions
- _Hackathon Marketing Manager_ - Introducing hackathon to external communities (local and global)
- _Judges x4_
- _Mentors x4_

### Budget
- Hackathon organizing tool: $3,000 USD
- Hackathon support roles (paid monthly)
-- Lead - 1000 USD in AKT (over 3 months)
-- Outreach Manager - 1000 USD in AKT (over 3 months)
-- Event Manager - 1000 USD in AKT (over 3 months)
-- Marketing Manager - 1000 USD in AKT
- Rewards: 100,000 AKT (split among 3 winners)
- Swag: $2,000 USD
- Paid Sponsor Tracks 
-- (Paid in native token by sponsors + $USD support)

Total community pool request in AKT (current prices as of Feb 27th, 2023: $.49)
112,000 AKT + 10% overage = 11,200AKT
**Total Request: 123,200 AKT**

Looking forward to reviewing Feedback.

**Comment by rodri-r** [comment]
I also believe that a hackathon is a great way of bringing attention, new developers and users to Akash. 

I am interested in becoming the Lead role for organizing this event and would like to share some of my ideas.  

The goals and timeline proposed by Adam seem reasonable, as well as reachable to me. These goals would definitely help to grow Akash and I will work towards achieving every one of them. 

I only differ a bit on the roles and budget sections. The following are my suggestions:

- Get rid of the Outreach Manager role and use this budget on two more Regional Marketing Managers and an extra incentive for the Lead role, as this role would take over the responsibilities of the Outreach Manager. 
--Examples of regions for marketing managers - continents - one manager for America, one manager for Europe and one manager for Asia.) 

**Budget proposal**

- Hackathon support roles (Total over three months/entire event)
-- Lead - $4000 USD in AKT
-- Event Manager - $3000 USD in AKT
-- Regional Marketing Managers - $3000 USD in AKT - ($1000 USD in AKT per regional manager)
-- Judges x4 and Mentors x4 - $800 USD in AKT - $100 USD in AKT each - (Taken from the +10% overage proposed by Adam)  
- Hackathon organizing tool, Rewards, Swag and other budgets proposed by Adam stay the same.

-----

As the Lead role, I would have no problems taking over the proposed responsibilities of an Outreach Manager.

I have experience working as a freelance developer, with support and marketing.  I have experience working as a team member, as well as leading groups.  I have experience with Akash and I am very aware that the main goal of this event is to promote Akash, as well as to encourage projects that benefit the network.  

I have participated in hackathons and I know how these events flow. I already have some options and ideas for organizing tools, mentoring sessions, judges and marketing. 

The Lead role can easily spend 20 hrs per week working on the organization of this event, which I am ready to work.  Being 100% fluent in English and Spanish, I would have no problem reaching out to potential partners, sponsors or communicating with the team and participants. 

The Lead role must be available at different hours of the day, as this role will have to interact with people from different parts of the world. I am used to working remotely and have no problem with scheduling calls at different hours of the day. 

As lead role, I will help participants before, during and after the hackathon. I will also create the dedicated hackathon website (suggested by Adam in the hackathon strategy doc) and host it on Akash.  

As mentioned before, my experience will allow me to achieve the proposed timeline/tasks, as well as working together with everyone that gets involved. 

My interest in Akash, tech and programming, encourages me to work towards achieving the proposed goals, as well  as making this hackathon fun and interesting.

**Reply by CaptainSplince** [reply]
Thanks for the feedback Rodri!

Love that your interested in leading the hackathon. I think roles will be selected once the final budget is approved via governance. Choice of roles can be debated openly in the Hackathon-WG discord channel.

As for your budget suggestion. My thoughts below:

Hackathon support roles (Total over three months/entire event)
-- Lead - $4000 USD in AKT 
>> This seems reasonable.
>> I don't think we would want to give "Outreach" to the lead role. They will be busy getting content prepared for distro, prepping the hackathon tooling, organizing each roles day to day, providing hackathon updates weekly. ect. Outreach IMO should be it's own role and responsibilities.

-- Event Manager - $3000 USD in AKT
>> This seems reasonable.

-- Regional Marketing Managers - $3000 USD in AKT - ($1000 USD in AKT per regional manager)
>> I don't fully understand this role. How will you be able to track their support with the hackathon? What will they be doing exactly?
The Outreach manager role would essentially do the role of "marketing manager." Since they would help onboard different sponsors (for sponsored tracks), those sponsors will then promote the hackathon to their communities, thus reaching a greater audience. It's kind of a two for one role.

-- Judges x4 and Mentors x4 - $800 USD in AKT - $100 USD in AKT each - (Taken from the +10% overage proposed by Adam)
>> Judges and mentors would not be paid. The reason why is because sponsored tracks could have members of those sponsor teams provide support via mentorship and judging as part of their sponsored involvement. We also have Akash Insiders that may be interested in volunteering their time to mentor or perhaps judge projects.

**Comment by rodri-r** [comment]
Thank you for the reply too!

-----

**Here are my views on why I think we can get rid of the "Outreach Manager" role:**

- This role's duties can be combined between the Regional Marketing Managers and the Lead role.
- Marketing managers would also outreach to potential Cosmos, Web3 and blockchain related projects, startups and companies. They would set up a call/meeting between these potential partners and the lead role to complete the deal.  
- Each marketing manager would be required to set up at least three meetings between potential partners and the lead role. These partners/sponsors need to have minimum requirements too; like you mention, these sponsors would promote the hackathon to their communities, so they must have something going on already and they cannot be any startup or small project without a community.  
-- Some Insiders from Brazil and Africa have already shown interest in the Marketing Manager role (In  the Hackathon call and in Discord channel). They probably already know other Cosmos leaders, local developers, Web3/blockchain related startups and companies, which could be interested in participating in this event. 
- Regional marketing managers would need to create marketing materials in different languages and share them both online and locally. Proof of this would be shared with lead role and everyone.  We potentially already have English, Spanish and Portuguese covered, (maybe French too), which will help us reach a larger audience. 
- In my opinion, having three regional marketing managers and the lead role reaching out to potential partners, would open up more opportunities for this hackathon, rather than a single person who might only consider partners that they are familiar with.  

----- 

**Judges and Mentors**

My idea on judges, is to contact some Cosmos leaders and/or "influencers".  They would most likely be more willing to participate with an AKT incentive and for their time, as I plan to host an online judging/award event and ceremony.  One or two judges would be from the Akash core dev team, as winning projects must benefit Akash and core devs know what is best for the network. 

My idea on incentivising mentors is mainly for their time.  While I do agree that most partners and sponsors would already have mentors and that they would participate in this event without an incentive, there might be some smaller projects that do not have a team that can become mentors, so we would need to have Akash Insiders or core dev team members become mentors and they would spend a few hours working on this. (If there are insiders that volunteer to become mentors, we can forget about this).

**Reply by iamGeorgePro** [reply]
Thank you Rodri for your suggestions. 

Imagine you're planning a road trip with your friends. You have a group of people with different skills and strengths - one is great at navigating, another is an excellent driver, and another is fantastic at finding fun places to stop along the way. Each person has a specific role that they excel at, and together, you're able to plan an incredible trip.

Now, imagine that your friend suggests that you combine all these roles into one person - the driver should also navigate and find fun stops along the way. It sounds like a good idea, but as you start driving, you realize that it's not working out as well as you'd hoped. The driver is too busy trying to find the next turn to focus on finding fun stops, and as a result, you miss out on some great opportunities.

In the same way, combining the outreach manager role with the regional marketing managers and lead role might sound like a good idea on paper, but in practice, it might not work out as well as we hope. Each role has its own specific strengths and focus, and by combining them, we might miss out on key opportunities.

For example, the outreach manager likely has connections and relationships with potential partners that the marketing managers and lead role might not be aware of. These relationships could be the key to securing valuable partnerships and sponsors that we might miss out if we don't have a dedicated individuals for each important role.

Additionally, by adding outreach and partnership responsibilities to the marketing managers and lead role, we might be overburdening them and spreading their focus too thin. This could lead to inefficiencies and potential burnout, which could negatively impact the success of the hackathon.

In conclusion, while it might seem like combining roles is a good idea, it's important to remember that each role has its own specific strengths and focus. By having individuals for each important role we can ensure that each critical aspect of the hackathon is given the necessary attention and expertise, and that we don't miss out on key opportunities.

> Thank you for the reply too!
> 
> **Here are my views on why I think we can get rid of the "Outreach Manager" role:**
> 
> * This role's duties can be combined between the Regional Marketing Managers and the Lead role.
> * Marketing managers would also outreach to potential Cosmos, Web3 and blockchain related projects, startups and companies. They would set up a call/meeting between these potential partners and the lead role to complete the deal.
> * Each marketing manager would be required to set up at least three meetings between potential partners and the lead role. These partners/sponsors need to have minimum requirements too; like you mention, these sponsors would promote the hackathon to their communities, so they must have something going on already and they cannot be any startup or small project without a community.
>   -- Some Insiders from Brazil and Africa have already shown interest in the Marketing Manager role (In  the Hackathon call and in Discord channel). They probably already know other Cosmos leaders, local developers, Web3/blockchain related startups and companies, which could be interested in participating in this event.
> * Regional marketing managers would need to create marketing materials in different languages and share them both online and locally. Proof of this would be shared with lead role and everyone.  We potentially already have English, Spanish and Portuguese covered, (maybe French too), which will help us reach a larger audience.
> * In my opinion, having three regional marketing managers and the lead role reaching out to potential partners, would open up more opportunities for this hackathon, rather than a single person who might only consider partners that they are familiar with.
> 
> **Judges and Mentors**
> 
> My idea on judges, is to contact some Cosmos leaders and/or "influencers". They would most likely be more willing to participate with an AKT incentive and for their time, as I plan to host an online judging/award event and ceremony. One or two judges would be from the Akash core dev team, as winning projects must benefit Akash and core devs know what is best for the network.
> 
> My idea on incentivising mentors is mainly for their time. While I do agree that most partners and sponsors would already have mentors and that they would participate in this event without an incentive, there might be some smaller projects that do not have a team that can become mentors, so we would need to have Akash Insiders or core dev team members become mentors and they would spend a few hours working on this. (If there are insiders that volunteer to become mentors, we can forget about this).

**Reply by rodri-r** [reply]
Hey George, thanks for sharing your ideas too!

In my opinion, in this case it would be better to divide the outreach manager tasks beteween the lead and several regional marketing managers, for several reasons:

- The sponsor tracks are a key part of a hackathon, so what if a single  "Hackathon Outreach Manager" fails to onboard partners of the quality that we should be focusing on? What if for any reason the partners this person thought they could onboard or that they already have a relationship with, could not make it to the hackathon or are not interested in Akash?

- I don't think that dividing the outreach tasks between four people would put a high burden on any of us or make our work any harder. I believe we could achieve more as a team and onboarding different partners/sponsors together, which we can then select from and invite the best to participate in our event.

**Reply by CaptainSplince** [reply]
Thanks for the feedback @Rodri-R & @iamGeorgePro!

To keep things simple and costs down, how do the following roles sound?

- Hackathon Lead (manages team, hackathon tooling, graphics, timeline, budget, swag, prize distribution, reports, ect.)
- Hackathon Community Manager (manages hacker events and hacker support)
- Hackathon Marketing Manager (leads hackathon promotion and partner sponsored tracks)

(Additional support but not included in roles will be provided by members of the Overclock Labs team)

**Comment by rodri-r** [comment]
Thanks @CaptainSplince 

I do like these roles and tasks per role specifications. It's simple and well organized. 

Having some Overclock Labs team helping out is great too!

**Comment by cloud-j-luna** [comment]
Great proposal @CaptainSplince !
I'd like to also propose to sync this hackathon with the work being done in the Client Libraries.
Participants would be encoraged (either through more points or extra rewards) to use the libraries in their solutions.
Such an approach would push developers into building tools and products that would enhance the Akash Ecosystem and interact with the Akash Network.

Project ideas:
* Provider migration tools
* HA controller
* Utility APIs
* Wallets
* Automations
...

For more information on client libraries visit [https://github.com/akash-network/community/tree/main/wg-client-libraries](https://github.com/akash-network/community/tree/main/wg-client-libraries)

**Reply by CaptainSplince** [reply]
Totally agree. If this proposal passes, next step is to define the projects/tracks hackers will have the opportunity to work on!

**Comment by SophoclesSummer** [comment]
This proposal was approved on 4/4/2023 by a 90% majority. 
https://www.mintscan.io/akash/proposals/186
